src/GameLogic.cpp

```diff
@@ -21,30 +22,43 @@ PlayerActor GameLogic::getPlayer(){
   return player;
 }
 
+void GameLogic::EPressed(){
+	Etracker++;
+	//std::cout << Etracker;
+}
+
 void GameLogic::upPressed(float dt){
-	if(!detectCollisionUp(dt))
-		player.moveUp(dt);
+	if (Etracker%4 == 0){
```

Comment:
like here, as long as the player is not using the dialogue and the tracker is at the limit or above it, then the player can move, otherwise freeze